Jenkinsfile

```diff
@@ -0,0 +1,76 @@
+#!groovy
+gcpProject =  "bw-prod-platform0"
+baseImageExist = false
+
+def doesBaseImageExist(String imageVersion) {
+    /*
+        Originally, the docker file would create a new debian image and install
+        all the dependencies needed to build solr, to save time, I've separated
+        the two "stages" into separate docker files, with this, I then do a check
+        to see if the ant-base image exists in our gcr, if it does, then it'll skip
+        building it which should cut down on processing time
+    */
+    try {
+        img = docker.image("${gcpProject}/ant-base:${imageVersion}").withRun { c ->
+            sh 'ant --help'
+        }
+        return true
+    } catch (Exception e) {
+        return false
+    }
+}
+
+def pushImage(String imageID, String imageVersion) {
+    withEnv(["DOCKER_BUILDKIT=1"]) {
+        img = docker.image(imageID)
+        docker.withRegistry("https://eu.gcr.io", "gcr:${gcpProject}") {
+            img.push(imageVersion)
+        }
+    }
+}
+
+node('docker') {
+    stage ('Checkout') {
+        checkout([
+            $class: 'GitSCM',
+            branches: scm.branches,
+            doGenerateSubmoduleConfigurations: scm.doGenerateSubmoduleConfigurations,
+            extensions: scm.extensions + [[$class: 'CleanCheckout']],
+            userRemoteConfigs: scm.userRemoteConfigs
+        ])
+
+        versions = readJSON file: "versions.json"
+        version = versions["version"]
+        major_version = versions["major_version"]
+    }
+
+    stage ('Build images') {
+        baseImageExist = doesBaseImageExist(major_version)
+        if ( baseImageExist ) {
+            echo "Base image exists, using that"
+            withEnv(["DOCKER_BUILDKIT=1"]) {
+                sh """ docker build \
+                    -f Dockerfile \
+                    --build-arg REPO=${gcpProject} \
+                    --build-arg MAJOR_VERSION=${major_version} \
+                    -t ${gcpProject}/bw-lucene-solr:${version} .
+                """
+                pushImage("${gcpProject}/bw-lucene-solr:${version}", version)
```

Comment:
I would recommend only pushing if you're on the master branch. Otherwise you can overwrite an image just by raising a PR. See: https://github.com/BrandwatchLtd/bw-airflow/blob/db6c449f050d4a507005f7feb96af74e1028fc0f/Jenkinsfile#L47

Reply:
The branches don't work like that for solr, you have;
```
[apache branches]
branch_x_y(_z)
i.e.
branch_7_7_2
branch_8_4

[bw branches (patches/custom stuff applied)]
bw_branch_x_y(_z)
ie.e.
bw_branch_7_7_2
bw_branch_8_4
```

Master on bw side, is massively out of date, and on apache is the most recent, often un-released versions of solr

Reply:
In that case you can check if you're on a pull request with:
```groovy
if (!env.CHANGE_ID) {
    // not a PR, so can push the images
}

Reply:
https://jenkins.brandwatch.net/job/BrandwatchLtd/job/bw-lucene-solr/pipeline-syntax/globals#env

Reply:
I just thought, I could push another tag which is the build number, that would probs be safest

Reply:
Yeah I think a lot of bw repos use `build.${build-number}` or something to that effect